File: clients/rust/model/src/bullet.rs
use crate::*;
#[derive(Clone, Debug, trans::Trans)]
pub struct Bullet {
    pub weapon_type: WeaponType,
    pub unit_id: i32,
    pub player_id: i32,
    pub position: Vec2F64,
    pub velocity: Vec2F64,
    pub damage: i32,
    pub size: f64,
    pub explosion_params: Option<ExplosionParams>,
}

File: clients/haskell/src/RAIC/Model/Bullet.hs
module RAIC.Model.Bullet (Bullet) where

import           GHC.Generics               (Generic)
import           RAIC.Model.ExplosionParams (ExplosionParams)
import           RAIC.Model.Vec2Double      (Vec2Double)
import           RAIC.Model.WeaponType      (WeaponType)
import           RAIC.Utils.Trans           (Trans)

data Bullet = Bullet {
  weapon_type      :: WeaponType,
  unit_id          :: Int,
  player_id        :: Int,
  position         :: Vec2Double,
  velocity         :: Vec2Double,
  damage           :: Int,
  size             :: Double,
  explosion_params :: Maybe ExplosionParams
} deriving (Generic, Show)

instance Trans Bullet
